packages/core/__tests__/src/utils/framework.test.ts

```diff
@@ -60,17 +59,12 @@ describe("framework.detectFramework", () => {
         })
       );
 
-      const fallback =
-        value === Frameworks.gatsby ? Frameworks.gatsby : Frameworks.vanillajs;
-
-      const wanted = FrameworkUtils.isValidFramework(value as Frameworks)
+      // if framework is not supported, vanillajs should always be sent back.
+      const wanted = FrameworkUtils.isFrameworkSupported(value as Frameworks)
         ? value
-        : fallback;
+        : Frameworks.vanillajs;
```

Comment:
it is not a good practice to write conditions in the tests because that means we need a tests to test it :p. 

The solution here : splitting more the test into different tests for define frameworks / isFrameworkSupported.